EIPS/eip-2390.md

```diff
@@ -0,0 +1,314 @@
+            }
+        }
+
+        // pointer is now node representing the resolution of the query geohash.
+        // DFS until all addresses found or ret[] is full.
+        // Do not use recursion because this is a blockchain...
+        uint[] memory indexes_to_visit = new uint[](TREE_VISITATION_QUEUESZ);
+        indexes_to_visit[0] = pointer;
+        uint front_i = 0;
+        uint back_i = 1;
+
+        while(front_i != back_i) {
+            Node memory cur_node = geomap[node][indexes_to_visit[front_i]];
+            front_i ++;
+
+            // if not a leaf node...
+            if (cur_node.data == address(0)) {
+                // visit all the chilin's
+                for(uint i=0; i<cur_node.children.length; i++) {
+                    // only visit valid children
+                    if (cur_node.children[i] != 0) {
+                        assert(back_i < TREE_VISITATION_QUEUESZ);
+                        indexes_to_visit[back_i] = cur_node.children[i];
+                        back_i ++;
+
+                    }
+                }
+            } else {
+                ret[ret_i] = cur_node.data;
+                ret_i ++;
+                if (ret_i > MAX_ADDR_RETURNS) break;
+            }
+        }
+
+        return ret;
+    }
+
+    // when setting, geohash must be precise to 8 digits.
+    function setGeoAddr(bytes32 node, string calldata geohash, address addr) external authorised(node) {
+        bytes32(node); // single node georesolver ignores node
+        require(bytes(geohash).length == 8); // 8 characters = +-1.9 meter resolution
+
+        // create root node if not yet created
+        if (geomap[node].length == 0) {
+            geomap[node].push( Node({
+                data: address(0),
+                parent: 0,
+                children: new uint256[](32)
+            }));
+        }
+
+        // walk into the geomap data structure
+        uint pointer = 0; // not actual pointer but index into geomap
+        for(uint i=0; i < bytes(geohash).length; i++) {
+
+            uint8 c = chartobase32(bytes(geohash)[i]);
+
+            if (geomap[node][pointer].children[c] == 0) {
+                // nothing found for this geohash.
+                // we need to create a path to the leaf
+                geomap[node].push( Node({
+                    data: address(0),
+                    parent: pointer,
+                    children: new uint256[](32)
+                }));
+                geomap[node][pointer].children[c] = geomap[node].length - 1;
+            }
+            pointer = geomap[node][pointer].children[c];
+        }
+
+        Node storage cur_node = geomap[node][pointer]; // storage = get reference
+        cur_node.data = addr;
+
+        emit GeoENSRecordChanged(node, geohash, addr);
+    }
+
+    function supportsInterface(bytes4 interfaceID) public pure returns (bool) {
+        return interfaceID == ERC2390 || super.supportsInterface(interfaceID);
+    }
+}
+```
+# References
```

Comment:
```suggestion
## References
```

This isn't an official section, and while extra sections are not strictly prohibited they are discouraged.  I recommend just inlining these since they are pretty short, and only referenced once.  Alternatively, consider including them in the section they are referenced in.